Migrate the CLI section to Nextra

jmintuitive opened the pull request:
### Description

Migrates the CLI section of the docs from Docusaurus to Nextra. 

Steps involved:
1. Revising long numbered sections to use \<Steps\>
2. Fixing links (internal to the section and in other migrated files - this will need to be touched up by future migration work as some files haven't been migrated yet)
3. Organize the information so that it shows in the same way as before.
4. Add more context to code snippets using Nextra features.
5. Revising old tutorials to work as `.mdx` files.
6. Clicking every link in the CLI sections to verify they go to the proper destination.

Also PR also fixed a bad link in `working-with-move-contracts` that was missed previously and fixed a small typo in the Nextra README.

The recommended way to review this PR is to pull down this branch locally, then run the local development environment and click through every CLI page to verify that the output is clear and bug-free. (Due to the Steps component, some of the sections are a bit harder to read in pure markdown format as there is no numbering inside the steps and the "rich preview" on GitHub doesn't display those numbered steps as headers). 

### Checklist

- Do all Lints pass?
  - [] Have you ran `pnpm spellcheck`?
  - [X] Have you ran `pnpm fmt`?
  - [X] Have you ran `pnpm lint`?

jmintuitive commented:
@gregnazario @hariria Please review the migrated CLI docs when you have a chance :)

hariria commented on apps/nextra/pages/en/build/cli/install-cli/install-cli-mac.mdx:
````diff
@@ -0,0 +1,36 @@
+### Ensure you have `brew` installed https://brew.sh/.
+### Open a new terminal and enter the following commands.
+
+```bash
+    brew update
+    brew install aptos
+```
````
For some reason there's an indent for the codeblocks (https://developer-docs-nextra-git-fork-jmintuitive-mig-294fb4-aptoslabs.vercel.app/en/build/cli/install-cli/install-cli-mac). 

Can we do a find and replace for all instances of ```bash under the `cli` folder. And instead make it 

````
```bash filename="Terminal"
````

Note: windows instructions for the CLI would be

````
```bash filename="Powershell"
````

gregnazario replied:
For Windows, do this instead, since bash isn't powershell

````
```powershell filename="Terminal"
````

hariria replied:
I'm not sure if 

````
```powershell
````

has syntax highlighting but down to try!

hariria commented on apps/docusaurus/docs/tools/aptos-cli/use-cli/working-with-move-contracts.md:
```diff
@@ -4,7 +4,7 @@ title: "Working With Move Contracts"
 # Working With Move Contracts
 
-The Aptos CLI is mostly used to compile, test, and formally verify Move contracts. If you have not installed the Aptos CLI yet, you can do so by following the steps here [Install the Aptos CLI (In-between Page)](https://www.notion.so/Install-the-Aptos-CLI-In-between-Page-9b923ca3507c4ffd8291f905ba621b95?pvs=21).
+The Aptos CLI is mostly used to compile, test, and formally verify Move contracts. If you have not installed the Aptos CLI yet, you can do so by following the steps here [Install the Aptos CLI](../install-cli/index.md).
```
Did you mean to edit this?

jmintuitive replied:
Yeah, that's a mistake from earlier - that new relative link is correct.

hariria commented on apps/nextra/README.md:
````diff
@@ -13,7 +13,7 @@ the local dev server:
 ```
 
-2. Navigate back to `netra`
+2. Navigate back to `nextra`
````
tyty

hariria commented on apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts/arguments-in-json-tutorial.mdx:
````diff
@@ -0,0 +1,576 @@
+
+Start by mining a vanity address for Ace, who will deploy the package:
+
+```bash filename=Command
+aptos key generate \
+    --vanity-prefix 0xace \
+    --output-file ace.key
+```
+
+
+<details>
+<summary>Output</summary>
+
+```bash filename="Output"
+{
+  "Result": {
+    "Account Address:": "0xacef1b9b7d4ab208b99fed60746d18dcd74865edb7eb3c3f1428233988e4ba46",
+    "PublicKey Path": "ace.key.pub",
+    "PrivateKey Path": "ace.key"
+  }
+}
+```
````
What do you think about changing these to 

````
```bash filename="Terminal"
````

hariria commented on apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts/arguments-in-json-tutorial.mdx:
````diff
@@ -0,0 +1,576 @@
+
+Start by mining a vanity address for Ace, who will deploy the package:
+
+```bash filename=Command
````
I think this needs quotes

hariria commented on apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts/arguments-in-json-tutorial.mdx:
````diff
@@ -0,0 +1,576 @@
+
+Now publish the package under Ace's account:
+
+```bash filename=Command
````
```suggestion
```bash filename="Command"
```

hariria commented on apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts/arguments-in-json-tutorial.mdx:
````diff
@@ -0,0 +1,576 @@
+
+First compile the package (this will compile the script):
+
+```bash filename=Compilation
````
```suggestion
```bash filename="Compilation"
```

gregnazario commented on apps/nextra/pages/en/build/cli/cli-configuration.mdx:
```diff
@@ -0,0 +1,97 @@
+
+This makes using the CLI easier and more secure as you will not be forced to repeatedly copy addresses or private keys.
+
+<Callout type="warning" emoji="⚠️">
+If you still need to install the CLI, follow [these steps](install-cli/install-cli-specific-version.mdx).
+</Callout>
+
+1. Run `aptos init` and follow the instructions in the command line.
+  <Callout type="info" emoji="ℹ️">
+   To use default settings, you can provide no input and just press “Enter”. For example:
+   </Callout>
```
Btw, unless Andrew asked directly, I don't think the emoji is needed

gregnazario commented on apps/nextra/pages/en/build/cli/install-cli/install-cli-mac.mdx:
````diff
@@ -0,0 +1,36 @@
+```
+</Steps>
+
+<Callout type="warning" emoji="⚠️">
````
Let's go forward with this, and we'll replace all of these with shared components

hariria replied:
I think it'll render correctly without the `emoji` property

jmintuitive replied:
It does indeed :)

gregnazario reviewed (approved):
We will probably need to do passes anyways over these visually on the new website

hariria commented on apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts.mdx:
```diff
@@ -0,0 +1,243 @@
+
+The Aptos CLI is mostly used to compile, test, and formally verify Move contracts. If you have not installed the Aptos CLI yet, you can do so by following the steps here [Install the Aptos CLI](../install-cli.mdx).
+
+You can jump to sections here:
+
+1. [Compiling Move](#1-compiling-move)
+2. [Unit Testing Move Contracts](#2-unit-testing-move-contracts)
+3. [Generating Test Coverage Reports](#3-generating-test-coverage-reports)
+4. [Publishing Move Contracts](#4-publishing-move-contracts)
+5. [Running Published Contracts](#5-running-published-contracts)
+6. [(Optional) Formally Verifying Move Scripts](#6-optional-formally-verifying-move-scripts)
```
I think we can get rid of this since we have a Table Of Contents (TOC)

hariria commented on apps/nextra/pages/en/build/cli/use-cli/running-a-local-network.mdx:
```diff
@@ -0,0 +1,188 @@
+1. **No rate limits:** You can interact with hosted services like the Node API, Indexer API, and faucet with no rate-limits to speed up testing.
+2. **Reproducibility:** You can set up specific on-chain scenarios and restart the network from scratch at any point to return to a clean slate.
+3. **High availability**: The Aptos devnet and testnet networks are periodically upgraded, during which time they can be unavailable. Local development networks are also always available even if you have no internet access.
+
```
Can we add a `<br />` here. Feels like it needs a bit of space (https://developer-docs-nextra-git-fork-jmintuitive-mig-294fb4-aptoslabs.vercel.app/en/build/cli/use-cli/running-a-local-network)

jmintuitive replied:
Agreed, looks much better :)

hariria reviewed (approved):
Nice job! A few nits but otherwise LGTM

jmintuitive commented:
@gregnazario @hariria The merge conflicts were from the `localnet` naming changes. From that I noticed that `running-a-local-network.mdx` (in the CLI section) and `local-development-network.mdx` in the Node guide are the same content - so I updated the content to match. 

A better solution would be to use the same file for both, but I didn't see how to do that with Nextra from a quick google. So I just updated the content in the Nodes side for now.
